src/blockdata/mimblewimble.rs

```diff
@@ -0,0 +1,272 @@
+// MimbleWimble transaction.
```

Comment:
Only skimming this file, since it's only partly implemented so far. When you're ready to finish this and start writing tests, let me know. I can generate some test vectors for you.

Reply:
I'm ready to write tests.

Reply:
@DavidBurkett ping?

Reply:
@webwarrior-ws Sure, what format do you need them in? Something like this?

Tx with 2 pegouts:
```
{
  "hex": "0200000000080000011819a9e1c5d2be3ff51d4041b281e355e9fe86e2389f69e79a81d73b32e873f7f056438d0d702760f98d20c851a7ce32ea9a995f53ebb460ebdb3264634849910101483f1e4f7f9e36824fd61e562fc07d48b3451e823d5f6afea4d5d8374b997f7c086b85e49b94ba898bbc8120e9253d3a8b2bf9b465ff7b8ace694e341a6fd6f18e033d7d91b44bef189813559e734245974b56a73ad6e483fd2f0cfeca9dde4a5c5d03db2bb489c1e9ed0104f1c8064766f3999f427eaeb20bebed973e57954cd4a66c0179db7cc9f7a9f83c5173f129143f478178436fb6ca6bc54076062a4905f8d5a91b04e0bb6af88eacb030323eea73c5678f27f26787864546a6f4fc92888f350108269e60eeaa812b4e776cf6a6e22b213679021bc1532e7541ccb606b73c254168020eab5a7b8897be8eeaff91b8899c492a9a39a02d6c0211f97aa7381c78671f850355cdb42339a479bd9b8506d6bcdc351aa769a5b63cca9b9b1bac9337cd06e27e010236ccd266c02cae55ed3be49fb239154c95714975a9576eb4dc5e58f7d524b753bb62273ae98ea4405c80b535f6d53702916acc1cc07ab048a62acb5a6bacf7f6a30d5c4979a17d6b05d9e0fb9040fc74e110fac309c1942d72ea169c65e0cf9bfd7c7ffb406ba2b49b062a7705731fde8c7f22cf05d02a20710c20145fc0d90bd6d10283098ecd07bced51df89f586309d8523ca27d9f26f9b61a5de20e8adcbe2ff4e7b238a6f40f22cf67d0f5b893e07998d2e9778e2f7091e9b10aab174a8d8ce64eedd90b3a24606c7c69d45eb94ff386c01705adf57f436bcf93048dac2b094c071d38eebf05046a7e63d033156f0a1a32214d28c426782da2f4f6b19605583e64f54a6c068caa8d14d3edee2221808182919c87ecd0c7c2f794e4b9db828cfeea6be8974cd93ecb0a0f9ae5d3e949bb15d64c8ed84f0ea8b447856217b42dff7d9fac4e5a949ed358a1dfd2d330a51acd41df1b7c8f1952143d210258fb925a60fb9b61ea23c6704bdcf35394d4d44f8dc2c20f9141c3567bf01ccc1b9232d2aa6fb7969afa36c029bb8230eef3a189664db9524ede6d8ac00e38686cb26c02c77556b1a1699d695f9300578171b3d8c84a7f2ae2f5de7a147fe3e1fec8bddf52b7843ae1b1f7d5ec372869496d4932060a978cb1d92c273002230f10fba5a8d086ca7681710bb955e77c8f87b71e90af2caa5d94e406e928925f76eb22663d8ce24af4d5af8cc6bcf62652327a129d6a6e51e7da944aeead34b2449965b792730aeb8d6e32b970f83b17fbb3cbc2d144490784e55e53e8912d8ea85d6e5f5046ebe6aa407e325ac3291a091f446d2ac63d9ab28382584b95856d641f5ae6fcca92b8ff1bc328b63995280ab66bd598dab66abac17691906e910ab8c2e35e8d34118ca9c2556cb9d2e9afab98f4f98d449b8ed3dd733d11ea0bfb31b55d35750d2bd22f123025b7ad7f9194cee14a11a617da84c067b5bc273c4c8a9f5ff42f5db9de68a2567bb1f1b5a522a2f5d9dcd80ffbd8396f7965b2aade9bbf12d918740c27c64dda7df2113d65fbb9710e77dd1ecd9c7bb9eb4bd5bf01d301f905a562a7cd7c9b967b405882511d1acde3457d22918f57e64f58fdb01159a540282dbea9300160014cfc8a91900d66314f40d54fe85b55e27cd2c505a80edb4c900160014eac5b26e7ac2eb063d216328e890a75eb7e446de021a6c99c8996aef706e13f6cced269369f73974ff283d46a4f31663952add67650940e5f34a5a341457ac8b197e0205df71510c0e0c0c317b584d3377ac1e4b0a185a460c2501617c413dfdb3a131dbd4c3904d43a66f7cd62f9596f6df5a49d76efdac29f967f0b4763aa6f5f9078a9ee5e38c9ae1b8cd558b4c49b32c31d7474c00000000",
  "tx": {
    "nLockTime": 0,
    "nVersion": [
      2
    ],
    "vin": "[]",
    "vout": "[]",
    "witness": "CTxWitness()",
    "hogex": false,
    "mweb_tx": {
      "body": {
        "mweb_inputs": [
          {
            "features": 1,
            "input_pk": "03db2bb489c1e9ed0104f1c8064766f3999f427eaeb20bebed973e57954cd4a66c",
            "output_commit": "086b85e49b94ba898bbc8120e9253d3a8b2bf9b465ff7b8ace694e341a6fd6f18e",
            "output_id": "483f1e4f7f9e36824fd61e562fc07d48b3451e823d5f6afea4d5d8374b997f7c",
            "output_pk": "033d7d91b44bef189813559e734245974b56a73ad6e483fd2f0cfeca9dde4a5c5d",
            "sig": "0179db7cc9f7a9f83c5173f129143f478178436fb6ca6bc54076062a4905f8d5a91b04e0bb6af88eacb030323eea73c5678f27f26787864546a6f4fc92888f35"
          }
        ],
        "mweb_kernels": [
          {
            "features": 21,
            "fee": 3540,
            "kernel_excess": "0940e5f34a5a341457ac8b197e0205df71510c0e0c0c317b584d3377ac1e4b0a18",
            "pegout_count": 2,
            "pegouts": [
              {
                "amount": 1000000000,
                "script": "0014cfc8a91900d66314f40d54fe85b55e27cd2c505a"
              },
              {
                "amount": 500000000,
                "script": "0014eac5b26e7ac2eb063d216328e890a75eb7e446de"
              }
            ],
            "sig": "5a460c2501617c413dfdb3a131dbd4c3904d43a66f7cd62f9596f6df5a49d76efdac29f967f0b4763aa6f5f9078a9ee5e38c9ae1b8cd558b4c49b32c31d7474c",
            "stealth_excess": "021a6c99c8996aef706e13f6cced269369f73974ff283d46a4f31663952add6765"
          }
        ],
        "mweb_outputs": [
          {
            "features": 1,
            "key_exchange_pk": "0236ccd266c02cae55ed3be49fb239154c95714975a9576eb4dc5e58f7d524b753",
            "masked_value": 6647493983704065890,
            "masked_value_nonce": "80b535f6d53702916acc1cc07ab048a6",
            "output_commit": "08269e60eeaa812b4e776cf6a6e22b213679021bc1532e7541ccb606b73c254168",
            "rangeproof": "2acb5a6bacf7f6a30d5c4979a17d6b05d9e0fb9040fc74e110fac309c1942d72ea169c65e0cf9bfd7c7ffb406ba2b49b062a7705731fde8c7f22cf05d02a20710c20145fc0d90bd6d10283098ecd07bced51df89f586309d8523ca27d9f26f9b61a5de20e8adcbe2ff4e7b238a6f40f22cf67d0f5b893e07998d2e9778e2f7091e9b10aab174a8d8ce64eedd90b3a24606c7c69d45eb94ff386c01705adf57f436bcf93048dac2b094c071d38eebf05046a7e63d033156f0a1a32214d28c426782da2f4f6b19605583e64f54a6c068caa8d14d3edee2221808182919c87ecd0c7c2f794e4b9db828cfeea6be8974cd93ecb0a0f9ae5d3e949bb15d64c8ed84f0ea8b447856217b42dff7d9fac4e5a949ed358a1dfd2d330a51acd41df1b7c8f1952143d210258fb925a60fb9b61ea23c6704bdcf35394d4d44f8dc2c20f9141c3567bf01ccc1b9232d2aa6fb7969afa36c029bb8230eef3a189664db9524ede6d8ac00e38686cb26c02c77556b1a1699d695f9300578171b3d8c84a7f2ae2f5de7a147fe3e1fec8bddf52b7843ae1b1f7d5ec372869496d4932060a978cb1d92c273002230f10fba5a8d086ca7681710bb955e77c8f87b71e90af2caa5d94e406e928925f76eb22663d8ce24af4d5af8cc6bcf62652327a129d6a6e51e7da944aeead34b2449965b792730aeb8d6e32b970f83b17fbb3cbc2d144490784e55e53e8912d8ea85d6e5f5046ebe6aa407e325ac3291a091f446d2ac63d9ab28382584b95856d641f5ae6fcca92b8ff1bc328b63995280ab66bd598dab66abac17691906e910ab8c2e35e8d34118ca9c2556cb9d2e9afab98f4f98d449b8ed3dd733d11ea0bfb31b55d35750d2bd22f123025b7ad7f9194cee14a11a617da84c067b5bc273c4c8a9f5ff42f5db9de68a2567bb1f1b5a522a2f5d9dcd80ffbd8396f7965b2a",
            "receiver_pk": "0355cdb42339a479bd9b8506d6bcdc351aa769a5b63cca9b9b1bac9337cd06e27e",
            "sender_pk": "020eab5a7b8897be8eeaff91b8899c492a9a39a02d6c0211f97aa7381c78671f85",
            "sig": "ade9bbf12d918740c27c64dda7df2113d65fbb9710e77dd1ecd9c7bb9eb4bd5bf01d301f905a562a7cd7c9b967b405882511d1acde3457d22918f57e64f58fdb",
            "view_tag": 187
          }
        ]
      },
      "kernel_offset": "1819a9e1c5d2be3ff51d4041b281e355e9fe86e2389f69e79a81d73b32e873f7",
      "stealth_offset": "f056438d0d702760f98d20c851a7ce32ea9a995f53ebb460ebdb326463484991"
    }
  }
}
```

Reply:
@DavidBurkett added a test based on your example. Maybe another test based on HogEx transaction?

Reply:
@DavidBurkett can this PR get merged first and then we contribute a new PR for HogEx?

Reply:
Taking longer to generate these than I hoped. Here's a couple more that should cover the basic flows. We can have a separate PR after to cover the various different input, output, and kernel features that can be set.

Pegin Tx:
```
{
  "hex": "0200000000090136e6c16875c3086a9fb5a70871029449adba9ebad76777ae5a9da246642837720000000000feffffff02a44dcd1d000000001600140891f455bb6050f01a10830d2e08fbb424657ff434d29a3b00000000225920660306b11402a8a3c2f13ff9bd1a6f7e1aaa3f811fb72d498bf8e0b11f9eb80e0247304402202fb3baae1fb68afd2049213153263ee137949a2c4c97350f5264c1c67ab4c97a02206c73857f5f43bbd1a718c84e04ecf2c6c0ebb1ad9e71fae6a6d2b718d21996a40121023ce0d60ddcb8532d727eba7d30af806f78f699d4f9d5adbf689f6c51056cfc6f0145b1970abcac1bd4573f5e0c560559d7ff667c9313bd03818c8ed4bf911a45f93af29c489fc4ea4e590986e795497af0f2e934b90eb7396b800264b4121ceaa70001099fec72900b4d2f4456af758ede240cde125b6fd60a24bb50330189c23bfc0ee4023996ae313b6c9a7c31a8ef1bf848309d2eefa6c00661b8d6779195154aca9b12032d58dde1dd55ee45738142d514e0e991284992be52536d11278453914b85ea310103d1277935595d55855d23383ac76e588b965cd2cc0507078bc3864712b29e1f70c97f0f0ac2aa9c6f77aa0ba2a63410236144f8189c85edc8e70475302bee96140914f21e24c326f2388be033398083528842b060a8aa2269ff918088d355dc2b1d44a0ab637f9b5321e64d09b70db7723a42873014690cd53f03d729ad91b7eb3e8bf8290f8087cfeacf635f28e4d79096805f1a29a43dc03cf1f8dd45eb2adb88b11b3e47f28a6d98004296e543d6292b73d533394357418c885f4b5a77f179483c95b6af67b04e34a727dedcd355c729e00bc5924e63d3a79a1dfce8e2866f0e8085d119e4ad37093390a0295f90ad38fc805e562fb064978b59a09465b65f95128d58bf658129387b1e5f7941f0ccc6c79940d0c6ba885910e97940b074d5f837d0b1bb5c988d92df4d4b88f0573d3da32daafce8afb247d4148e70a8f5552552c5cb07f084e9e4d89362cf6da1fefcfa351b9171f5cd9e884316d6e1950d109c07a46d1a4ec78e59143b60e1618ab44c5c8ec6ca45498521ce4c49da68688a806510501fde9c2e0ecaf88c11ee139152ea675f886f259788940180ea82c07b43b05ba90449cf8bfcbaed7391fd8e0e17307ccec89cfdc7af5d35dae246031cad8b96b38f0e3e83bddb5f2773e4dd5283070852400d17caca73cf2c7f7446473c175c56595e9d15b91b0165c8e6bbc2b5ae0768541519a3f443ca913979d94bd33cd7fbe5c8b313dad81c785f5e0044b7eb89f0535a45cf536caeea3bec6403a2e14cff3bdb10351f65dbc6e34b24ed010c0a46c43a196b828d424a550fd2963a2e7a1950557af3431c1ec40a07cad2ef01ab18a803226a767f365bd1a847fb253ac17c3d90746a084e6323f30bc9248a35024ce62be1c42d1bd5e72698f6e5f7ed6d44ed322fb02d0b511add03285cdcfeddfa8649bea60dda801f66eda05505b0817878c0685f977899a400d4b7f52db0f9b1dda45b059519200583f7ed8f4b45fd3f36ca8b5a6121035ee6260e0fbebf88b5846d23ca900a74b93c661d067d3648b1d8d4d17bb4976b48c18a6045ee0bd8ba198622adea2eada2caaa1be6648bfe897f05ad14757e0f42dac095c359a73a716d7f0cfcc16a0001138f3482dbeaa334034b5b5b5d5d78c6282cd4b4dd4c940e4a3e5e1ff5d30c314669482f0e92acd21e08796dda8095e7fc07130c2262b3a39bdc8ecf2560313f05f522de99f25817df57317c36b334f1db15e0bfeeb595a29b6f50dfdff33f5d246576c774284d257a8371bb59eedd4c63c6fd8a9c6c5f84d1b861c91b38b91e362664e46f7d93058c1800000000",
  "tx": {
    "hogex": false,
    "mweb_tx": {
      "body": {
        "mweb_inputs": [],
        "mweb_kernels": [
          {
            "features": 19,
            "fee": 2100,
            "kernel_excess": "08796dda8095e7fc07130c2262b3a39bdc8ecf2560313f05f522de99f25817df57",
            "pegin_amount": 1000002100,
            "sig": "317c36b334f1db15e0bfeeb595a29b6f50dfdff33f5d246576c774284d257a8371bb59eedd4c63c6fd8a9c6c5f84d1b861c91b38b91e362664e46f7d93058c18",
            "stealth_excess": "034b5b5b5d5d78c6282cd4b4dd4c940e4a3e5e1ff5d30c314669482f0e92acd21e"
          }
        ],
        "mweb_outputs": [
          {
            "features": 1,
            "key_exchange_pk": "03d1277935595d55855d23383ac76e588b965cd2cc0507078bc3864712b29e1f70",
            "masked_value": 8606269670142119807,
            "masked_value_nonce": "aa0ba2a63410236144f8189c85edc8e7",
            "output_commit": "099fec72900b4d2f4456af758ede240cde125b6fd60a24bb50330189c23bfc0ee4",
            "rangeproof": "0475302bee96140914f21e24c326f2388be033398083528842b060a8aa2269ff918088d355dc2b1d44a0ab637f9b5321e64d09b70db7723a42873014690cd53f03d729ad91b7eb3e8bf8290f8087cfeacf635f28e4d79096805f1a29a43dc03cf1f8dd45eb2adb88b11b3e47f28a6d98004296e543d6292b73d533394357418c885f4b5a77f179483c95b6af67b04e34a727dedcd355c729e00bc5924e63d3a79a1dfce8e2866f0e8085d119e4ad37093390a0295f90ad38fc805e562fb064978b59a09465b65f95128d58bf658129387b1e5f7941f0ccc6c79940d0c6ba885910e97940b074d5f837d0b1bb5c988d92df4d4b88f0573d3da32daafce8afb247d4148e70a8f5552552c5cb07f084e9e4d89362cf6da1fefcfa351b9171f5cd9e884316d6e1950d109c07a46d1a4ec78e59143b60e1618ab44c5c8ec6ca45498521ce4c49da68688a806510501fde9c2e0ecaf88c11ee139152ea675f886f259788940180ea82c07b43b05ba90449cf8bfcbaed7391fd8e0e17307ccec89cfdc7af5d35dae246031cad8b96b38f0e3e83bddb5f2773e4dd5283070852400d17caca73cf2c7f7446473c175c56595e9d15b91b0165c8e6bbc2b5ae0768541519a3f443ca913979d94bd33cd7fbe5c8b313dad81c785f5e0044b7eb89f0535a45cf536caeea3bec6403a2e14cff3bdb10351f65dbc6e34b24ed010c0a46c43a196b828d424a550fd2963a2e7a1950557af3431c1ec40a07cad2ef01ab18a803226a767f365bd1a847fb253ac17c3d90746a084e6323f30bc9248a35024ce62be1c42d1bd5e72698f6e5f7ed6d44ed322fb02d0b511add03285cdcfeddfa8649bea60dda801f66eda05505b0817878c0685f977899a400d4b7f52db0f9b1dda45b059519200583f7ed8f4b45fd3f36ca8b5a6121035ee6260e0fbebf88b5846d23ca900a74",
            "receiver_pk": "032d58dde1dd55ee45738142d514e0e991284992be52536d11278453914b85ea31",
            "sender_pk": "023996ae313b6c9a7c31a8ef1bf848309d2eefa6c00661b8d6779195154aca9b12",
            "sig": "b93c661d067d3648b1d8d4d17bb4976b48c18a6045ee0bd8ba198622adea2eada2caaa1be6648bfe897f05ad14757e0f42dac095c359a73a716d7f0cfcc16a00",
            "view_tag": 201
          }
        ]
      },
      "kernel_offset": "45b1970abcac1bd4573f5e0c560559d7ff667c9313bd03818c8ed4bf911a45f9",
      "stealth_offset": "3af29c489fc4ea4e590986e795497af0f2e934b90eb7396b800264b4121ceaa7"
    },
    "nLockTime": 0,
    "nVersion": [
      2
    ],
    "vin": "[CTxIn(prevout=COutPoint(hash=7237286446a29d5aae7767d7ba9ebaad4994027108a7b59f6a08c37568c1e636 n=0) scriptSig= nSequence=4294967294)]",
    "vout": "[CTxOut(nValue=4.99994020 scriptPubKey=00140891f455bb6050f01a10830d2e08fbb424657ff4), CTxOut(nValue=10.00002100 scriptPubKey=5920660306b11402a8a3c2f13ff9bd1a6f7e1aaa3f811fb72d498bf8e0b11f9eb80e)]",
    "witness": "CTxWitness(CScriptWitness(304402202fb3baae1fb68afd2049213153263ee137949a2c4c97350f5264c1c67ab4c97a02206c73857f5f43bbd1a718c84e04ecf2c6c0ebb1ad9e71fae6a6d2b718d21996a401,023ce0d60ddcb8532d727eba7d30af806f78f699d4f9d5adbf689f6c51056cfc6f))"
  }
}
```

Block with HogEx:
```

{
  "hex": "0000002079e60c7cbe960e964576f1ca68e0c5db432516c6bf2bc9fe1f98d12baaa2f9ce142940e0325f34e42af3c03095045d249b89ef91ce6c32e9f3c58666eca1eb530ee6bb65ffff7f200000000003020000000001010000000000000000000000000000000000000000000000000000000000000000ffffffff0402b30100ffffffff02dc93814a000000001976a9144ff785b8221dc206314ca12e65773a876dff30ff88ac0000000000000000266a24aa21a9ed69c2bd004db215de53ecba7dbd6704c69c79091e026352527f6acb93a2954519012000000000000000000000000000000000000000000000000000000000000000000000000002000000000101e875d6a28e1c2bb8dec7c644d0c13e771c3342e0043db395b454f52e3776d8530000000000feffffff02a44dcd1d0000000016001420be6a6f419926d269318e82c52dac12ae0c7d0a34d29a3b00000000225920a9cf87cc73baa78ccd0632ef9723792f0fb68b7fb3e5648ec7acf220b402c4760247304402204b596d3f4a7b38775a5d7f66ab853c515e8922f5adcdf45c550ebd7be2bb39f202200b325e415afc07711771efab7195f646b18ebaedec766114be651024ee2eabef012103cbb7fd473df7adf22f5fb00aba8e5c25f23a183cfc5565c9959c498e127a56f60000000002000000000802639a293622cc1069bc04ac31a35a5abbb1afb31e73b785c08152b1e668a570ea0000000000fffffffffd4903b0030fda98ea5b69d5cd52bfab71190baea7a454d9d01163f71dd965330100000000ffffffff0100ab904100000000225820c3ba0e85455458ab40eb2e210fd5fd4ab2c969f265c3d37bfbb3cf74eda3c4d90000000000018233a822eb3e28db055248a0ae4510052bc24381c6c0323efa93bd745e6715e425a7745c9ac2352031762feccc8ba9b21353eb988f5f82e17943dbbb836753bf08e2af2edc674154ff129d9e826727ada0828d3ba480924bbed84bf6dcae2e1f1db29e984f060b4233e1a34afb44ae006ffd7ade9abb05b3c78dad8029d696bf72f7c0ecbf5563ed14f264fda787033cc5f5270d8c17e49633b98c11908cdabf282d020100010871fe35144c1c69a2f669d0ba86788b881e586adcf4f467c59dfde31bbf2279ef02093961ce9a6be06eca61b0491b6a5cfcb19babf3d616482057a9b96933d12bbb0333c3e2213250cfa1741083d6ebfbcdbe40008fc8c6a58ca4234adf77bb4844580102a980d8bd1eb6cd20c549ac6667541b2f98822ba801cb06aaa996646881ce2f0a9cda0b7dfe62dd414ad212e0ee248e77a87960e7c7afa05470221fb94e2163a60084b741f6ab99e7fcc0e38088e0d30a15aa00365cdbe57894d4f1ebbc3645b06b4ab23b9634d769f88f5c188889510b63522e0ed514e9ea6c090c48cd3042686b6531c06c893c1d52ab35e440bffc68a4d6512cbe0ffe316be4f81ea4ff7412050c31fd48605c835944bf0c7d9f1e5641d3f6dbe0470621e90c7590e130ba95614768bc0ae588b68f331481cf53c322d50b21f43fb2b3869aad3c073f37700bf1b008959bd4ef8ce75da8e6ea940e7df5fcc773add1f444eed906d7532ce6d0c7e5f8ffbfe850fa1dd0ae3cd57a8bae56c7b67204109bb5b0e486be9df8c85ff8dc912f16de82181082750770ebfd0c830901aba039fe3d37e0233c6e2a57b744888b2b0d4a7bce0930d2cc57d51a0c9540597970311943c485988c76e5985d996bc5877e3619205141db4bd65f9f896ff31ec267ca8416ad979e9c78f5de921503a6fece2acee085510f01793e1e0e285d8ededc6bb58e3ef6b40319c5ae6a0b67e9e677b30d9c42ab11dfd25fa9cd9e567ed08bc959868980e8652201aa87fdc69b451c4a2e41317c728fae78dfd224403beafa8cb1634f7b1dd5ad5934be2b8a5cb7f0b2e04873aa2521c577990d7a39e6c4285634e553500ed8712790625b762238739bbe10525ca571147f15d074289368dc9164f402ebd415df479e7ba31e2a6c4994327fd47802723c94e1c6007653fd5ab36fff92401f2804c36a81d0f90d98c0b4d781bfa7d7e8c2e93313336a69a6c955a68a412373af664e1d9d383083ccf236ae5296d82bc4abe353f07faf23be12a10182581647e2fbb834db4e46c29854b7f18f52ed2a0ca55d5617cf6fa662654eba0d90fae318aa28c06981cb60685f3362d5f932eb5c2e66ea11230a293ad76d16d2ca5d218d204ffe3c5e220586aa9ec209be55a970366f830710e46a05f5605b5a73e63227db7314b8e8446fa933655ace36770e9170ab69fd06ce87a8d0e227b4652eefdabb1e523db877d10601347382348fb314387eca7fe41e0281b9f6432a7e876b5501138f3482dbeaa33402b0b948f32e91675452647396563882109c42ff8a9ea81e73966e197fed82d4a6086ef5b161faeb8bbc87f6b538fc399480d5733db5929fa9eea503b177e8b79da56af4a676f878fda7a4bfb0a46b8afb04528f71b2122a7aee9f3f1ae3044786932dcc06d6fae3be55160aad5f9562c7c34ae8d4a42c3ed117e54e73af565d85d4",
  "block": {
    "header": {
      "bits": 545259519,
      "merkle_root": "53eba1ec6686c5f3e9326cce91ef899b245d049530c0f32ae4345f32e0402914",
      "nonce": 0,
      "previous_block_hash": "cef9a2aa2bd1981ffec92bbfc6162543dbc5e068caf17645960e96be7c0ce679",
      "timestamp": 1706812942,
      "version": 536870912
    },
    "mweb": {
      "header": {
        "height": 435,
        "kernel_offset": 111924016465549674428331415742132999106345418936417393406889533896439265990814,
        "kernel_root": 102238159038465599408062154877722040558041103120382444304238933397197807049844,
        "leafset_root": 80563140825409524756254123105579764346219688760559158327985735385936866061999,
        "num_kernels": 1,
        "num_txos": 2,
        "output_root": 75603193230014258331018886200535704949729004944145275982452245501332937188008,
        "stealth_offset": 20426076194461720485449487382082433739467438835476895562233376256852968336576
      },
      "inputs": [],
      "kernels": [
        {
          "features": 19,
          "fee": 2100,
          "kernel_excess": "086ef5b161faeb8bbc87f6b538fc399480d5733db5929fa9eea503b177e8b79da5",
          "pegin_amount": 1000002100,
          "sig": "6af4a676f878fda7a4bfb0a46b8afb04528f71b2122a7aee9f3f1ae3044786932dcc06d6fae3be55160aad5f9562c7c34ae8d4a42c3ed117e54e73af565d85d4",
          "stealth_excess": "02b0b948f32e91675452647396563882109c42ff8a9ea81e73966e197fed82d4a6"
        }
      ],
      "outputs": [
        {
          "features": 1,
          "key_exchange_pk": "02a980d8bd1eb6cd20c549ac6667541b2f98822ba801cb06aaa996646881ce2f0a",
          "masked_value": 5350801249539001306,
          "masked_value_nonce": "d212e0ee248e77a87960e7c7afa05470",
          "output_commit": "0871fe35144c1c69a2f669d0ba86788b881e586adcf4f467c59dfde31bbf2279ef",
          "rangeproof": "221fb94e2163a60084b741f6ab99e7fcc0e38088e0d30a15aa00365cdbe57894d4f1ebbc3645b06b4ab23b9634d769f88f5c188889510b63522e0ed514e9ea6c090c48cd3042686b6531c06c893c1d52ab35e440bffc68a4d6512cbe0ffe316be4f81ea4ff7412050c31fd48605c835944bf0c7d9f1e5641d3f6dbe0470621e90c7590e130ba95614768bc0ae588b68f331481cf53c322d50b21f43fb2b3869aad3c073f37700bf1b008959bd4ef8ce75da8e6ea940e7df5fcc773add1f444eed906d7532ce6d0c7e5f8ffbfe850fa1dd0ae3cd57a8bae56c7b67204109bb5b0e486be9df8c85ff8dc912f16de82181082750770ebfd0c830901aba039fe3d37e0233c6e2a57b744888b2b0d4a7bce0930d2cc57d51a0c9540597970311943c485988c76e5985d996bc5877e3619205141db4bd65f9f896ff31ec267ca8416ad979e9c78f5de921503a6fece2acee085510f01793e1e0e285d8ededc6bb58e3ef6b40319c5ae6a0b67e9e677b30d9c42ab11dfd25fa9cd9e567ed08bc959868980e8652201aa87fdc69b451c4a2e41317c728fae78dfd224403beafa8cb1634f7b1dd5ad5934be2b8a5cb7f0b2e04873aa2521c577990d7a39e6c4285634e553500ed8712790625b762238739bbe10525ca571147f15d074289368dc9164f402ebd415df479e7ba31e2a6c4994327fd47802723c94e1c6007653fd5ab36fff92401f2804c36a81d0f90d98c0b4d781bfa7d7e8c2e93313336a69a6c955a68a412373af664e1d9d383083ccf236ae5296d82bc4abe353f07faf23be12a10182581647e2fbb834db4e46c29854b7f18f52ed2a0ca55d5617cf6fa662654eba0d90fae318aa28c06981cb60685f3362d5f932eb5c2e66ea11230a293ad76d16d2ca5d218d204ffe3c5e220586aa9ec209be55a970366f830710e46a05f5605b5a73e63227",
          "receiver_pk": "0333c3e2213250cfa1741083d6ebfbcdbe40008fc8c6a58ca4234adf77bb484458",
          "sender_pk": "02093961ce9a6be06eca61b0491b6a5cfcb19babf3d616482057a9b96933d12bbb",
          "sig": "db7314b8e8446fa933655ace36770e9170ab69fd06ce87a8d0e227b4652eefdabb1e523db877d10601347382348fb314387eca7fe41e0281b9f6432a7e876b55",
          "view_tag": 156
        }
      ]
    },
    "txs": [
      {
        "tx": {
          "hogex": false,
          "nLockTime": 0,
          "nVersion": 2,
          "vin": "[CTxIn(prevout=COutPoint(hash=0000000000000000000000000000000000000000000000000000000000000000 n=4294967295) scriptSig=02b30100 nSequence=4294967295)]",
          "vout": "[CTxOut(nValue=12.50005980 scriptPubKey=76a9144ff785b8221dc206314ca12e65773a876dff30ff88ac), CTxOut(nValue=0.00000000 scriptPubKey=6a24aa21a9ed69c2bd004db215de53ecba7dbd6704c69c79091e026352527f6acb93a2954519)]",
          "witness": "CTxWitness(CScriptWitness(0000000000000000000000000000000000000000000000000000000000000000))"
        }
      },
      {
        "tx": {
          "hogex": false,
          "nLockTime": 0,
          "nVersion": 2,
          "vin": "[CTxIn(prevout=COutPoint(hash=53d876372ef554b495b33d04e042331c773ec1d044c6c7deb82b1c8ea2d675e8 n=0) scriptSig= nSequence=4294967294)]",
          "vout": "[CTxOut(nValue=4.99994020 scriptPubKey=001420be6a6f419926d269318e82c52dac12ae0c7d0a), CTxOut(nValue=10.00002100 scriptPubKey=5920a9cf87cc73baa78ccd0632ef9723792f0fb68b7fb3e5648ec7acf220b402c476)]",
          "witness": "CTxWitness(CScriptWitness(304402204b596d3f4a7b38775a5d7f66ab853c515e8922f5adcdf45c550ebd7be2bb39f202200b325e415afc07711771efab7195f646b18ebaedec766114be651024ee2eabef01,03cbb7fd473df7adf22f5fb00aba8e5c25f23a183cfc5565c9959c498e127a56f6))"
        }
      },
      {
        "tx": {
          "hogex": true,
          "nLockTime": 0,
          "nVersion": 2,
          "vin": "[CTxIn(prevout=COutPoint(hash=ea70a568e6b15281c085b7731eb3afb1bb5a5aa331ac04bc6910cc2236299a63 n=0) scriptSig= nSequence=4294967295), CTxIn(prevout=COutPoint(hash=3365d91df76311d0d954a4a7ae0b1971abbf52cdd5695bea98da0f03b00349fd n=1) scriptSig= nSequence=4294967295)]",
          "vout": "[CTxOut(nValue=11.00000000 scriptPubKey=5820c3ba0e85455458ab40eb2e210fd5fd4ab2c969f265c3d37bfbb3cf74eda3c4d9)]",
          "witness": "CTxWitness()"
        }
      }
    ]
  }
}
```

Reply:
@DavidBurkett I've added tests. I also fixed block height decoding, now proper amount of bytes is consumed so no more skipping a byte when decoding MWEB block header.

Reply:
@DavidBurkett ping, can this go in now?